perceval/backends/core/mbox.py

```diff
@@ -84,23 +83,115 @@ def fetch(self, from_date=DEFAULT_DATETIME):
+
+        kwargs = {'from_date': from_date}
+        items = super().fetch("message", **kwargs)
+
+        return items
+
+    def fetch_items(self, **kwargs):
+        """Fetch the messages"""
+
+        from_date = kwargs['from_date']
+
         logger.info("Looking for messages from '%s' on '%s' since %s",
                     self.uri, self.dirpath, str(from_date))
 
-        mailing_list = MailingList(self.uri, self.dirpath)
+        self.client = MailingList(self.uri, self.dirpath)
 
-        messages = self._fetch_and_parse_messages(mailing_list, from_date)
+        messages = self._fetch_and_parse_messages(self.client, from_date)
 
         for message in messages:
             yield message
 
         logger.info("Fetch process completed")
 
+    @classmethod
+    def has_caching(cls):
+        """Returns whether it supports caching items on the fetch process.
+
+        :returns: this backend does not support items cache
+        """
+        return False
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend does not support items archive
+        """
+        return False
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a MBox item."""
+
+        return item[MBox.MESSAGE_ID_FIELD]
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MBox item.
+
+        The timestamp used is extracted from 'Date' field in its
+        several forms. This date is converted to UNIX timestamp
+        format.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = item[MBox.DATE_FIELD]
+        ts = str_to_datetime(ts)
+
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a MBox item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return 'message'
+
+    @staticmethod
+    def parse_mbox(filepath):
+        """Parse a mbox file.
+
+        This method parses a mbox file and returns an iterator of dictionaries.
+        Each one of this contains an email message.
+
+        :param filepath: path of the mbox to parse
+
+        :returns : generator of messages; each message is stored in a
+            dictionary of type `requests.structures.CaseInsensitiveDict`
+        """
+        mbox = _MBox(filepath, create=False)
+
+        for msg in mbox:
+            message = message_to_dict(msg)
+            yield message
+
+    def _init_client(self, from_archive=False):
+        """Init mailing list"""
+
+        return MailingList(self.uri, self.dirpath)
```

Comment:
This method it isn't used inside `MBox` so don't override it and keep it as in `Backend`.